aiohttp/http_writer.py

```diff
@@ -171,11 +171,26 @@ async def drain(self) -> None:
             await self._protocol._drain_helper()
 
 
+def _check_string(string: str) -> str:
```

Comment:
I'm not sure how I feel about a function which just returns the string unchanged, but it should atleast have a better name. Maybe `_checked_header()` or `_safe_header()`?

I'm also wondering if it's not better to check the values earlier, where the user sets them, rather than when they are being sent.

Reply:
I left it as `_check_string` because I use it to check status line as well. Not sure what would be a better name, maybe I should just add a docstring explaining what this function is intended for?

As for checking the values when user sets them - in this case I am not sure where in the codebase the valuation would take place - would veryfing the headers in `client.py` and `web_response.py` be enough or did you have another approach in mind?

I think it makes sense to check the values earlier, when they are being set. But I worry it would be easy to miss some case like for example when `charset`is passed (might be more cases like that). Checking the values when they are being sent seems more bulletproof, given that the headers may be mutated by some other code. Also I am wondering if this approach could potentially break existing code. Let me know what you think about it, but for now I would be more comfortable with checking the values when they are being sent.

Reply:
> I left it as `_check_string` because I use it to check status line as well.

I think the status line is still part of the head, so not sure I'm too bothered by that. But, do we even need to check the status line? I've not run through the code yet, but I'd have thought the status line does not involve any user input.

> As for checking the values when user sets them - in this case I am not sure where in the codebase the valuation would take place - would veryfing the headers in `client.py` and `web_response.py` be enough or did you have another approach in mind?

@webknjaz Do you think there's a better place for this check to live?

Reply:
I think you are right, there is no need to validate the status line. I changed the name of the function and removed validation for status line.

Reply:
So, just taking a look through the codebase, the alternative way I could imagine this working, is that `ClientRequest.headers` and `StreamResponse._headers` are both set to a `CIMultiDict`. If we changed this class to perform the check when a value is set in the dict, that could work. Though we'd need to also catch cases where the user passes in their own dict, so maybe it's not worth the complexity.